### PR TITLE
Feature/689 constant now listeners

### PR DESCRIPTION
**Pull request by LeoColman**

### CODE REVIEW

File: doc/extensions.md [hunk 1]
```diff
@@ -260,4 +260,36 @@ And with the listeners
     TimeZoneTestListener(TimeZone.getTimeZone(ZoneId.of("America/Sao_Paulo")))
   )
 
+```
+
+### Current instant listeners
+
+Sometimes you may want to user `java.time` classes' `now` static function for multiple reasons, such as setting the creation date of an entity
```
Comment (sksamuel) [inline doc/extensions.md]: Sometimes you may want to use the `now` static functions located in `java.time` ...

File: doc/extensions.md [hunk 2]
```diff
@@ -260,4 +260,36 @@ And with the listeners
...
+
+`data class MyEntity(creationDate: LocalDateTime = LocalDateTime.now())`.
+ 
+But what to do when you want to test that value? `now` will be different
+each time you call it!
+
+For that, KotlinTest provides `ConstantNowListener` and `withConstantNow` functions.
+
+While executing your code, your `now` will always be the value that you want to test against!!
```
Comment (sksamuel) [inline doc/extensions.md]: No need for !! i don't think.

File: kotlintest-extensions/src/test/kotlin/com/sksamuel/kt/extensions/time/ConstantNowExtensionsTests.kt [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  override fun afterSpecClass(spec: Spec, results: Map<TestCase, TestResult>) {
+    HijrahDate.now() shouldNotBeSameInstanceAs myNow
+    LocalDateTime.now() shouldNotBeSameInstanceAs myNow2
+  }
+  
+  override fun listeners() = listOf(
+          ConstantNowTestListener(myNow),
+          ConstantNowTestListener(myNow2)
+  )
+}
```
Comment (sksamuel) [inline kotlintest-extensions/src/test/kotlin/com/sksamuel/kt/extensions/time/ConstantNowExtensionsTests.kt:267]: Really good idea and clever implementation. We can put this in 3.3 if you like.

**Review by sksamuel — CHANGES_REQUESTED** [review]

File: kotlintest-extensions/src/test/kotlin/com/sksamuel/kt/extensions/time/ConstantNowExtensionsTests.kt (hunk 3 above)
Comment (LeoColman) [reply kotlintest-extensions/src/test/kotlin/com/sksamuel/kt/extensions/time/ConstantNowExtensionsTests.kt:267]: Sure!

My only issues with the implementation is that technically speaking, `Temporal` doesn't force any implementation to contain a static `now`. All `Temporal`s from Java do, but maybe a random temporal appears, and it won't work.

I said that in the docs, but idk if that's a concern.

I didn't want to create 17 implementations of 17 types with 17 ultra-similar executions

File: doc/extensions.md (hunk 1 above)
Comment (LeoColman) [reply doc/extensions.md]: much better. Brainfarted there

**Comment by LeoColman** [comment]
Alright! Merging this guy.

**Pull request merged**